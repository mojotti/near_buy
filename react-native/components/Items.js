'use strict';

import React from 'react';
import { Button,
    ListView,
    StyleSheet,
    Text,
    View } from 'react-native';
import { logout } from '../redux/actions/auth';
import { connect } from 'react-redux';
import { localhost,
    width } from "../src/static/constants";

export class Items extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            dataSource: new ListView.DataSource({
                rowHasChanged: (row1, row2) => row1 !== row2,
            }),
            loaded: false,
            data: ''
        };
    }

    static navigationOptions = {
        title: 'Your items',
        headerLeft: null
    };

    userLogout() {
        this.props.onLogout();
    }

    componentDidMount() {
        this.fetchData();
    }

    getHeaders() {
        let headers = new Headers();
        headers.append('Authorization', `Bearer ${this.props.token}`);
        return headers;
    }

    fetchData() {
        let url = `http://${localhost}:5000/api/v1.0/user/items`;
        fetch(url, {
            method: 'GET',
            headers: this.getHeaders()
        })
        .then((response) => response.json())
        .then((responseJson) => {
            this.handleAllItemsRequest(responseJson);
        })
        .catch((error) => {
        console.error(error);
        });
    }

    handleAllItemsRequest(responseJson) {
        if (responseJson.items === 'no items') {
        this.setState({
            loaded: true,
            data: responseJson.items,
        });
        } else {
            this.setState({
                dataSource: this.state.dataSource.cloneWithRows(responseJson.items),
                loaded: true,
                data: responseJson.items,
            });
        }
    }

    navigateToNewItem() {
        const { navigate } = this.props.navigation;
        navigate('NewItem');
    }

    render() {
        return (
            <View style={[styles.container]}>
                {this.renderUserData()}
                <Button
                    title='Add new item'
                    onPress={() => this.navigateToNewItem()}
                />
                <Button
                    onPress={() => this.userLogout()}
                    title='Logout'
                    style = {{margin: 15}}
                />
            </View>
        );
    }

    renderUserData() {
        if (this.state.loaded === false) {
            return (
                <Text style={[styles.infoText]}>
                    {'Loading user data...'}
                </Text>
            )
        }
        if (this.state.data === 'no items') {
            return (
                <Text style={[styles.infoText]}>
                    {`You don't have any items yet. You should start to sell some shit!`}
                </Text>
            );
        } else {
            return (
                <ListView
                    dataSource={this.state.dataSource}
                    renderRow={data => this.renderItem(data)}
                />
            );
        }
    }

    renderItem(item) {
        return (
            <View style={[styles.item, styles.separator]}>
                <Text>
                    {`Title: ${item.title}`}
                </Text>
                <Text>
                    {`Description: ${item.description}`}
                </Text>
                <Text>
                    {`Price: ${item.price} â‚¬`}
                </Text>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
    },
    item: {
        padding: 10,
        backgroundColor: 'azure',
        width: width,
    },
    separator: {
        marginBottom: 1,
        borderBottomWidth: 1,
        borderBottomColor: 'gainsboro',
    },
    infoText: {
        textAlign: 'center',
        padding: 10
    }
});

const mapStateToProps = (state, ownProps) => {
    return {
        username: state.auth.username,
        token: state.auth.token
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        onLogout: () => { dispatch(logout()); }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Items);
