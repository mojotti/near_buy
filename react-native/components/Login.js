'use strict';

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Alert,
    Button,
    Image,
    KeyboardAvoidingView,
    StyleSheet,
    Text,
    TextInput,
    View } from 'react-native';
import { login } from '../redux/actions/auth';
import { generateHeadersForBasicAuth,
    generateHashForRegistering } from '../src/networking';
import { localhost,
    loginText,
    registerText,
    widthWithThirtyPercentPadding} from '../src/static/constants';

export class Login extends Component {
    constructor (props) {
        super(props);
        this.state = {
            page: 'Login',
            username: '',
            password: '',
            email: '',
            textFocused: false
        };
    }

    togglePage () {
        let currentPage = this.getPageTitle(this.state);
        this.setState({ page: currentPage });
    }

    getPageTitle (state) {
        let { page } = state;
        return (page === 'Login') ? 'Sign up' : 'Login';
    }

    getHelperText (state) {
        let { page } = state;
        return (page === 'Login') ? loginText : registerText;
    }

    handleButtonPress(state) {
        const { page, username, password, email } = state;
        if(username === '' || password === '' || (page === 'Sign up' && email === ''))
        {
            Alert.alert('Invalid values', 'Please enter all the values.');
            return;
        }
        if (page === 'Login') {
            this.handleLoginRequest(state)
        } else {
            this.handleRegisteringRequest(state)
        }
    }

    handleRegisteringRequest (state) {
        const { username, password, email } = state;
        let ip_addr = `http://${localhost}:5000/api/v1.0/register`;
        fetch(ip_addr, {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                'user_info': generateHashForRegistering(username, password, email)
            })
        })
        .then((response) => response.json())
        .then((responseJson) => {
            if (responseJson.user_creation === 'success') {
                Alert.alert('User creation', 'User created successfully!');
                console.log(responseJson);
                this.setState({ page: 'Login' });
            } else {
                Alert.alert('User creation', 'User exists already!');
            }
        })
        .catch((error) => {
            console.log('error ', error);
            console.error(error);
        });
    }

    handleLoginRequest (state) {
        let ip_addr = `http://${localhost}:5000/api/v1.0/auth`;
        const { username, password } = state;
        fetch(ip_addr, {
            method: 'GET',
            headers: generateHeadersForBasicAuth(username, password)
        })
        .then((response) => response.json())
        .then((responseJson) => {
            console.log(responseJson);
            if (responseJson.username === username) {
                this.props.onLogin(username, responseJson.token);
            } else {
                Alert.alert('Try again, mate!', 'Invalid credentials.');
                this.setState({ password: '' });
            }
        })
        .catch((error) => {
            console.error(error);
        });
    }

    renderEmailInput() {
        if (this.state.page === 'Sign up') {
            return (
                <TextInput
                    placeholder='Email address'
                    autoCapitalize='none'
                    autoCorrect={false}
                    autoFocus={false}
                    keyboardType='email-address'
                    value={this.state.email}
                    onFocus={() => this.setState({textFocused: true})}
                    onChangeText={(text) => this.setState({ email: text })}
                />
            )
        }
    }

    renderLogoAndWelcomeText() {
        if (this.state.textFocused === false) {
            return (
                <View style={[styles.logoContainer]}>
                  <Text style={[styles.welcomeText]}>Welcome to NearBuy</Text>
                  <Image
                      resizeMode="contain"
                      style={styles.logo}
                      source={require('./../logo.png')}
                  />
                </View>
            )
        }
    }

    render () {
        return (
            <View style={[styles.container]}>
              <KeyboardAvoidingView
                  behavior='padding'
                  keyboardVerticalOffset={64}
              >
                  {this.renderLogoAndWelcomeText()}
                <View style={[styles.loginContainer]}>
                    {this.renderEmailInput()}
                  <View style={{margin: 4}}></View>
                  <TextInput
                      placeholder='Username'
                      autoCapitalize='none'
                      autoCorrect={false}
                      autoFocus={false}
                      onFocus={() => this.setState({textFocused: true})}
                      keyboardType='email-address'
                      value={this.state.username}
                      onChangeText={(text) => this.setState({ username: text })} />
                  <View style={{margin: 4}}></View>
                  <TextInput
                      placeholder='Password'
                      autoCapitalize='none'
                      autoCorrect={false}
                      secureTextEntry={true}
                      onFocus={() => this.setState({textFocused: true})}
                      value={this.state.password}
                      onChangeText={(text) => this.setState({ password: text })} />

                  <Text style={{fontSize: 27}}>{this.state.route}</Text>
                  <Button onPress={() => this.handleButtonPress(this.state)} title={this.state.page}/>
                  <View style={{marginTop: 40, flexDirection: 'row', justifyContent: 'center'}}>
                    <Text onPress={() => this.togglePage()} style={{fontSize: 16, color: 'blue'}}>
                        {this.getPageTitle(this.state)}
                    </Text>
                  </View>
                  <Text style={[styles.loginHint]}>{this.getHelperText(this.state)}</Text>
                </View>
              </KeyboardAvoidingView>
            </View>
        );
    }
}


const mapStateToProps = (state, ownProps) => {
    return {
        isLoggedIn: state.auth.isLoggedIn
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        onLogin: (username, token) => { dispatch(login(username, token)); },
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'space-around',
    },
    welcomeText: {
        fontSize: 24,
        textAlign: 'center',
        marginTop: 50
    },
    loginHint: {
        fontSize: 12,
        textAlign: 'center',
        marginTop: 10,
    },
    logo: {
        width: 350,
        height: 150,
    },
    loginContainer: {
        justifyContent: 'center',
        alignSelf: 'center',
        width: widthWithThirtyPercentPadding,
    },
    logoContainer: {
        justifyContent: 'center',
        alignSelf: 'stretch',
        padding: 0
    },
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
