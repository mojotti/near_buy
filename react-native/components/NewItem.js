'use strict';

import React, { Component } from 'react';
import store from '../redux';
import { Alert,
    Button,
    KeyboardAvoidingView,
    StyleSheet,
    TextInput,
    View } from 'react-native';
import { NavigationActions } from 'react-navigation';
import { localhost,
    widthWithThirtyPercentPadding} from "../src/static/constants";


export class NewItem extends Component {
    constructor(props) {
        super(props);
        this.state = {
            title: '',
            price: '',
            description: '',
        };
    }

    static navigationOptions = {
        title: 'Sell item',
    };

    getHeaders() {
        var headers = new Headers();
        headers.append('Content-Type', 'application/json');
        headers.append('Authorization', `Bearer ${store.getState().auth.token}`);
        return headers;
    }

    resetNavigationAndNavigateToRoute(targetRoute) {
        const resetAction = NavigationActions.reset({
            index: 0,
            actions: [
                NavigationActions.navigate({ routeName: targetRoute }),
            ],
        });
        this.props.navigation.dispatch(resetAction);
    }

    handleNewItemCreation () {
        const { title, price, description } = this.state;
        if (title == '' || price == '' || description == '') {
            Alert.alert('Invalid values', 'Enter at least price, title and description');
            return;
        }
        let url = `http://${localhost}:5000/api/v1.0/items`;
        fetch(url, {
            method: 'POST',
            body: JSON.stringify({
                'title': title,
                'price': price,
                'description': description
            }),
            headers: this.getHeaders()
        })
            .then((response) => response.json())
            .then((responseJson) => {
                console.log("pure " + responseJson.item.title);
                if (responseJson.item && responseJson.item.title === title) {
                    this.resetNavigationAndNavigateToRoute('Items');
                } else {
                    Alert.alert('Item creation failed', 'Something went wrong');
                }
            })
            .catch((error) => {
                console.error(error);
                return;
            });
    }

    onPriceChange(text) {
        let newText = '';
        let numbers = '0123456789';

        for (var i = 0; i < text.length; i++) {
            if ( numbers.indexOf(text[i]) > -1 ) {
                newText = newText + text[i];
            }
        }
        this.setState({price: newText})
    }

    render() {
        return (
            <View style={[styles.container]}>
                <TextInput
                    placeholder='Title'
                    autoCapitalize='sentences'
                    autoCorrect={false}
                    autoFocus={true}
                    maxLength = {20}
                    keyboardType='email-address'
                    value={this.state.title}
                    onChangeText={(text) => this.setState({ title: text })}
                    style={[styles.itemDetails]}
                />
                <TextInput
                    placeholder='Description'
                    autoCapitalize='sentences'
                    maxLength = {60}
                    autoCorrect={false}
                    autoFocus={false}
                    keyboardType='email-address'
                    value={this.state.description}
                    onChangeText={(text) => this.setState({ description: text })}
                    style={[styles.itemDetails]}
                />
                <KeyboardAvoidingView
                    behavior='padding'
                    keyboardVerticalOffset={164}
                >
                    <TextInput
                        placeholder='Price'
                        autoCorrect={false}
                        autoFocus={false}
                        keyboardType='numeric'
                        maxLength = {5}
                        value = {this.state.price}
                        onChangeText = {(text) => this.onPriceChange(text)}
                        style={[styles.itemDetails]}
                    />
                    <View style = {{paddingTop: 20}}></View>
                    <Button
                        onPress = {() => this.handleNewItemCreation()}
                        title = {'Submit item'}
                    />
                </KeyboardAvoidingView>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'space-around',
    },
    itemDetails: {
        width: widthWithThirtyPercentPadding,
        alignItems: 'center',
    }
});


export default NewItem;
