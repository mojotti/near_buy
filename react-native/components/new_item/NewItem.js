'use strict';

import React from 'react';
import store from '../../redux/index';
import {
  Alert,
  ScrollView,
  Text,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';
import ImagePicker from 'react-native-image-crop-picker';
import { NavigationActions } from 'react-navigation';
import {
  alertInvalidValuesNewItem,
  localhost,
} from '../../src/static/constants';
import { styles } from '../../src/static/styles/NewItemStyles';
import { ItemDetails } from './ItemDetails';

export class NewItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      price: '',
      description: '',
      mounted: false,
    };

    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
    this.handlePriceChange = this.handlePriceChange.bind(this);
  }

  static navigationOptions = {
    title: 'Add item',
  };

  componentDidMount() {
    // work around for testing issue, should be solved when RN version is updated
    this.setState({ mounted: true });
  }

  getHeaders() {
    const headers = new Headers();
    headers.append('Content-Type', 'application/json');
    headers.append('Authorization', `Bearer ${store.getState().auth.token}`);
    return headers;
  }

  resetNavigationAndNavigateToRoute(targetRoute) {
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName: targetRoute })],
    });
    this.props.navigation.dispatch(resetAction);
  }

  handleNewItemCreation() {
    const { title, price, description } = this.state;
    if (title === '' || price === '' || description === '') {
      Alert.alert(...alertInvalidValuesNewItem);
      return;
    }
    let url = `http://${localhost}:5000/api/v1.0/items`;
    fetch(url, {
      method: 'POST',
      body: JSON.stringify({
        title: title,
        price: price,
        description: description,
      }),
      headers: this.getHeaders(),
    })
      .then(response => response.json())
      .then(responseJson => {
        if (responseJson.item && responseJson.item.title === title) {
          this.resetNavigationAndNavigateToRoute('Items');
        } else {
          Alert.alert('Item creation failed', 'Something went wrong');
        }
      })
      .catch(error => {
        console.error(error);
      });
  }

  renderSubmitButton() {
    return (
      <TouchableHighlight
        id="SubmitButton"
        onPress={() => this.handleNewItemCreation()}
        style={styles.submitButton}
      >
        <Text style={styles.submitText}>{'Submit item'}</Text>
      </TouchableHighlight>
    );
  }

  showImagePicker() {
    return (
      <TouchableHighlight
        id="AddImageButton"
        onPress={() => this.openPicker()}
        style={styles.submitButton}
      >
        <Text style={styles.submitText}>{'Add photo'}</Text>
      </TouchableHighlight>
    );
  }

  openPicker() {
    ImagePicker.openPicker({
      width: 400,
      height: 400,
      cropping: true,
    }).then(image => {
      console.log(image);
    });
  }

  handleTitleChange(text) {
    this.setState({ title: text });
  }

  handleDescriptionChange(text) {
    this.setState({ description: text });
  }

  handlePriceChange(text) {
    this.setState({ price: text });
  }

  render() {
    const itemDetailsProps = {
      onTitleChange: this.handleTitleChange,
      onDescriptionChange: this.handleDescriptionChange,
      onPriceChange: this.handlePriceChange,
    };
    return (
      <ScrollView>
        <View style={styles.container}>
          <Text style={styles.headerText}>Item details</Text>
          <ItemDetails {...itemDetailsProps} />
          <Text style={styles.headerText}>Add pictures</Text>
          {this.showImagePicker()}
          {this.renderSubmitButton()}
        </View>
      </ScrollView>
    );
  }
}

export default NewItem;
