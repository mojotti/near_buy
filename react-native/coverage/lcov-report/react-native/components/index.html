<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for react-native/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> react-native/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.57% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>37/89</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.54% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>16/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.79% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>14/47</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.57% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>37/89</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Items.js"><a href="Items.js.html">Items.js</a></td>
	<td data-value="41.18" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.18" class="pct low">41.18%</td>
	<td data-value="34" class="abs low">14/34</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="20" class="abs low">6/20</td>
	<td data-value="41.18" class="pct low">41.18%</td>
	<td data-value="34" class="abs low">14/34</td>
	</tr>

<tr>
	<td class="file low" data-value="Login.js"><a href="Login.js.html">Login.js</a></td>
	<td data-value="41.82" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.82" class="pct low">41.82%</td>
	<td data-value="55" class="abs low">23/55</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="20" class="abs medium">15/20</td>
	<td data-value="29.63" class="pct low">29.63%</td>
	<td data-value="27" class="abs low">8/27</td>
	<td data-value="41.82" class="pct low">41.82%</td>
	<td data-value="55" class="abs low">23/55</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Oct 18 2017 21:32:17 GMT+0300 (EEST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
