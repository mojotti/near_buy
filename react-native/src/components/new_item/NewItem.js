'use strict';

import React from 'react';
import store from '../../../redux/index';
import {
  Alert,
  ScrollView,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import {
  alertInvalidValuesNewItem,
  localhost,
} from '../../static/constants';
import { styles } from '../../static/styles/NewItemStyles';
import { ItemDetails } from './ItemDetails';
import { ImageRow } from './ImageRow';

const UPPER_BUTTON_IDS = {
  leftButtonId: 0,
  rightButtonId: 1,
};

const LOWER_BUTTON_IDS = {
  leftButtonId: 2,
  rightButtonId: 3,
};

export class NewItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      price: '',
      description: '',
      images: [null, null, null, null],
    };

    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
    this.handlePriceChange = this.handlePriceChange.bind(this);
    this.setNewImage = this.setNewImage.bind(this);
  }

  static navigationOptions = {
    title: 'Add item',
  };

  getHeaders() {
    const headers = new Headers();
    headers.append('Content-Type', 'application/json');
    headers.append(
      'Authorization',
      `Bearer ${store.getState().authorizationReducer.token}`,
    );
    return headers;
  }

  resetNavigationAndNavigateToRoute(targetRoute) {
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName: targetRoute })],
    });
    this.props.navigation.dispatch(resetAction);
  }

  handleNewItemCreation() {
    const { title, price, description } = this.state;
    if (title === '' || price === '' || description === '') {
      Alert.alert(...alertInvalidValuesNewItem);
      return;
    }
    const url = `http://${localhost}:5000/api/v1.0/items`;
    fetch(url, {
      method: 'POST',
      body: JSON.stringify({
        title: title,
        price: price,
        description: description,
      }),
      headers: this.getHeaders(),
    })
      .then(response => response.json())
      .then(responseJson => {
        this.handleResponse(responseJson);
      })
      .catch(error => {
        console.error(error);
      });
  }

  handleResponse(responseJson) {
    if (responseJson.item && responseJson.item.title === this.state.title) {
      this.resetNavigationAndNavigateToRoute('Items');
    } else {
      Alert.alert('Item creation failed', 'Something went wrong');
    }
  }

  renderSubmitButton() {
    return (
      <TouchableHighlight
        id="SubmitButton"
        onPress={() => this.handleNewItemCreation()}
        style={styles.submitButton}
      >
        <Text style={styles.submitText}>{'Submit item'}</Text>
      </TouchableHighlight>
    );
  }

  setNewImage(image, id) {
    this.setState(prevState => {
      let images = prevState.images;
      images[id] = image;
      return images;
    });
  }

  handleTitleChange(text) {
    this.setState({ title: text });
  }

  handleDescriptionChange(text) {
    this.setState({ description: text });
  }

  handlePriceChange(text) {
    this.setState({ price: text });
  }

  render() {
    const itemDetailsProps = {
      onTitleChange: this.handleTitleChange,
      onDescriptionChange: this.handleDescriptionChange,
      onPriceChange: this.handlePriceChange,
    };

    const firstImageRowProps = {
      onImageSelected: this.setNewImage,
      ...UPPER_BUTTON_IDS,
      images: this.state.images.slice(0, 2),
    };

    const secondImageRowProps = {
      onImageSelected: this.setNewImage,
      ...LOWER_BUTTON_IDS,
      images: this.state.images.slice(2),
    };

    return (
      <ScrollView>
        <View style={styles.container}>
          <Text style={styles.headerText}>Item details</Text>
          <ItemDetails {...itemDetailsProps} />
          <Text style={styles.headerText}>Add pictures</Text>
          <ImageRow {...firstImageRowProps} style={styles.imageRowMargin} />
          <ImageRow {...secondImageRowProps} style={styles.imageRowMargin} />
          {this.renderSubmitButton()}
        </View>
      </ScrollView>
    );
  }
}

export default NewItem;
