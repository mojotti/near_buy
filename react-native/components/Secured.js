import React, { Component } from 'react';
import { Platform, StatusBar } from 'react-native';
import { StackNavigator } from 'react-navigation';
import Items from './Items.js';
import NewItem from './new_item/NewItem';

const App = StackNavigator({
  Items: { screen: Items },
  NewItem: { screen: NewItem },
});

class Secured extends Component {
  render() {
    return <App />;
  }
}

export default Secured;
