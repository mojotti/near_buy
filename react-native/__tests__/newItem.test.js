import React from 'react';
import renderer from 'react-test-renderer';

import { NewItem } from '../components/NewItem';

describe('New Item', () => {
    it('renders correctly', () => {
        const newItem = renderer.create(<NewItem />);

        expect(newItem.toJSON()).toMatchSnapshot();
    });
});