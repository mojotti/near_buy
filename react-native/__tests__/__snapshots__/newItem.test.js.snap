// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Item renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "marginTop": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "stretch",
            "fontFamily": "San Francisco",
            "fontSize": 22,
            "marginBottom": 15,
            "marginLeft": 56.25,
            "textAlign": "left",
          }
        }
      >
        Item details
      </Text>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "elevation": 1,
              "height": 80.03999999999999,
              "marginBottom": 10,
              "width": 637.5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="sentences"
            autoCorrect={false}
            autoFocus={undefined}
            keyboardType="email-address"
            maxLength={40}
            onChangeText={[Function]}
            placeholder="Title"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "marginLeft": 15,
                  "marginRight": 15,
                  "width": 525,
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "elevation": 1,
              "height": 266.8,
              "marginBottom": 10,
              "width": 637.5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="sentences"
            autoCorrect={false}
            keyboardType="email-address"
            maxLength={400}
            multiline={true}
            onChangeText={[Function]}
            placeholder="Description"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "marginLeft": 15,
                  "marginRight": 15,
                  "width": 525,
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "elevation": 1,
              "height": 80.03999999999999,
              "marginBottom": 10,
              "width": 637.5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            autoFocus={false}
            keyboardType="numeric"
            maxLength={5}
            onChangeText={[Function]}
            placeholder="Price"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "marginLeft": 15,
                  "marginRight": 15,
                  "width": 525,
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "stretch",
            "fontFamily": "San Francisco",
            "fontSize": 22,
            "marginBottom": 15,
            "marginLeft": 56.25,
            "textAlign": "left",
          }
        }
      >
        Add pictures
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4d4dff",
            "borderRadius": 30,
            "elevation": 1,
            "marginTop": 15,
            "paddingBottom": 8,
            "paddingTop": 8,
            "width": 525,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": "San Francisco",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Add photo
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4d4dff",
            "borderRadius": 30,
            "elevation": 1,
            "marginTop": 15,
            "paddingBottom": 8,
            "paddingTop": 8,
            "width": 525,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": "San Francisco",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Submit item
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
